### PR TITLE
FUSETOOLS2-223 - provide list of Camel K tasks

### PR DESCRIPTION
**Pull request by apupier**
filter out non-applicable task to context. For now, limiting when there
is no variable inside the "file" value

showing task defined for the file:
DefinedTaskInStartApacheCamelIntegration

filtering out when file does not corresponds:
filterOutIfFileDoesntApply

ideas for following iterations:
- evaluate also variables to determine if applicable
- provide options to create a new task directly

### CODE REVIEW

**Comment by apupier** [comment]
realizing that I need to update the readme and maybe a dd a note in the changelog

File: README.md [hunk 1]
```diff
@@ -182,14 +188,7 @@ For more information about configuration with ConfigMaps or Secrets, see [Config
...
 
-After you publish a new integration, it appears in the **Apache Camel K Integrations** view in the Side Bar of the Explorer activity:
-
-![Apache Camel K Integrations view](images/camelk-integration-view.jpg)
-
-When you add or remove file-based integrations in the Explorer view, it automatically refreshes the list.
-
-**Note:** Refreshing the view sometimes is delayed as pods start. You might need to wait a few seconds. Optionally, to manually refresh the list, click the **Refresh** button.
-
+After you publish a new integration, it appears in the **AReuse
```
Comment (bfitzpat) [inline README.md]: Is it "A Reuse" or is this a typo?

Comment (apupier) [reply README.md]: typo

Comment (apupier) [reply README.md]: even worse, a part of the text was removed. seems my fingers had troubles :s

**Comment by apupier** [comment]
a test is failing:

```
1) Kubectl integration watcher
       Check there is no loop for closing kubectl process:
     expected shareMessage to not have been called but was called twice
    shareMessage({ _channel: [Promise] {  }, name: "Apache Camel K" }, Error: file UnRelated.java does not exist: stat UnRelated.java: no such file or directory
) at Socket.<anonymous> (/home/travis/build/camel-tooling/vscode-camelk/out/src/kamel.js:9:8777)
    shareMessage({ _channel: [Promise] {  }, name: "Apache Camel K" }, W0318 08:38:58.891729    6231 loader.go:223] Config not found: /home/travis/.kube/config
Error: cannot get command client: could not locate a kubeconfig
) at Socket.<anonymous> (/home/travis/build/camel-tooling/vscode-camelk/out/src/kamel.js:9:8777)
  AssertError: expected shareMessage to not have been called but was called twice
      shareMessage({ _channel: [Promise] {  }, name: "Apache Camel K" }, Error: file UnRelated.java does not exist: stat UnRelated.java: no such file or directory
  ) at Socket.<anonymous> (out/src/kamel.js:9:8777)
  
      shareMessage({ _channel: [Promise] {  }, name: "Apache Camel K" }, W0318 08:38:58.891729    6231 loader.go:223] Config not found: /home/travis/.kube/config
  Error: cannot get command client: could not locate a kubeconfig
  ) at Socket.<anonymous> (out/src/kamel.js:9:8777)
  	at Object.fail (node_modules/sinon/lib/sinon/assert.js:107:21)
  	at failAssertion (node_modules/sinon/lib/sinon/assert.js:66:16)
  	at Object.assert.<computed> [as notCalled] (node_modules/sinon/lib/sinon/assert.js:92:13)
  	at Context.<anonymous> (out/src/test/suite/kubectlwatcher.test.js:50:26)
  	at Generator.next (<anonymous>)
  	at fulfilled (out/src/test/suite/kubectlwatcher.test.js:21:58)
```

looks more like tests are not really independent than a real issue but requires to investigate and fix

**Comment by apupier** [comment]
fixed the failing test. It was a side effect of using an unexisting file uri

File: README.md [hunk 2]
```diff
@@ -92,7 +92,11 @@ You can start a new Camel K integration with or without additional options such
...
 
-    Specify dependency details either by their camel-component artifact Id or by their Maven coordinates (group:artifact:version), with the option to specify more than one. For more information about configuration with Dependencies, see [Dependencies and Component Resolution](https://camel.apache.org/camel-k/latest/configuration/dependencies.html) in the Apache Camel-K documentation.
+    Specify dependency details either by their camel-component artifact Id or by their Maven coordinates (group:artifact:version), with the option to specify more than one. For more information about configuration with Dependencies, see [Dependencies and Component Resolution](https://camel.apache.org/camel-k/latest/configuration/dependencies.html) in the Apache Camel K documentation.
+
+* **Use a predefined Task - useful for multi-attributes deployment**
+
+    Allow to use a predefined Task of type "camel-k". It is useful when deploying an Apache Camel K integration which requires several parameters. To provide a task, see section [Creating a new Camel K Integration with multiple parameters](#Creating-a-new-Camel-K-Integration-with-multiple-parameters)
```
Comment (bfitzpat) [inline README.md]: I might change the language in this area slightly.

* **Define a Task for Easily Reproducible Multi-attribute Deployment**

Some types of deployments require more complex combinations of attributes. In these cases, defining a custom Task of type "camel-k" is useful. To create a task, see section [Creating a new Camel K Integration with multiple parameters](#Creating-a-new-Camel-K-Integration-with-multiple-parameters)

File: README.md [hunk 3]
```diff
@@ -102,9 +106,9 @@ To update the state of your currently deployed integrations, hover over the **Ap
 
 ![Apache Camel K Integrations view - Refresh](images/camelk-integrations-view-refresh-action.jpg)
 
-## Starting a new Camel K Integration with multiple parameters
+## Creating a new Camel K Integration with multiple parameters
 
-Using the right-click menu is the easiest way to start a new integration and works well for simple cases. When the Camel K integration requires more configuration, you can set that up using a Task.
+The right-click menu is providing by default several ways to start a new integration and works well for simple cases. When the Camel K integration requires more configuration, you can set that up using a Task.
```
Comment (bfitzpat) [inline README.md]: Might change this to:

Though the simple "Start Apache Camel K Integration" menu works well for simple cases, you can use a Task for more complex integrations.

File: README.md [hunk 4]
```diff
@@ -144,7 +148,9 @@ Once you've created your Camel K task, you can use auto-complete to explore the
 }
 ```
 
-When you're done, save the file and call command from palette `Tasks: Run Task`. You will see the command listed with the label that you provided previously.
+When you're done, you have two options to use it:
```
Comment (bfitzpat) [inline README.md]: Might change this to:

When you've defined your Task, you can use one of two options to execute it:

**Pull request merged**